### DISCUSSION TITLE
source .env No such file or directory

### DISCUSSION THREAD

**Discussion by malommey** [Cyfrin/foundry-full-course-cu#348] [Q&A] [answered]
```
when i ran the command in my project file directory source .env
```
i got this error

```
bash: .env: No such file or directory

```
here is my .env code

```
PRIVATE_KEY= 0xacXXXXXXX
RPC_URL= http://127.0.0.1:8545
```
please help

**Comment by ghost** [comment]
Hello @malommey the problem seems to be the spaces after the equal signs, remove them as shown in the following code snippet:

```bash
PRIVATE_KEY=0xacXXXXXXX
RPC_URL=http://127.0.0.1:8545
```

If the issue persist you can refer to discussion https://github.com/Cyfrin/foundry-full-course-f23/discussions/107 or https://github.com/Cyfrin/foundry-full-course-f23/discussions/109 to see other possible solutions.

**Reply by shikhar229169** [reply]
I think along with the error @pacelliv mentioned, the .env file actually doesn't exist.

**Reply by ghost** [reply]
Ups, I think this is the real issue. Thanks! I read too fast, but it's weird because apparently the file was created but probably wrote "env" without the dot?

**Reply by malommey** [reply]
@pacelliv i have repair the space between the value in my .env file and ran the command but didn’t solve it. i have also tried the solution provided in discussion #109 didn’t solve it

**Reply by shikhar229169** [reply]
According to me your .env file is in different directory.
Can you share a ss of your left pane of vs code, where the files and folders are shown.
Along with that share the screenshot of your command line.

**Reply by malommey** [reply]
@shikhar229169 okay here is the screenshot

![Screenshot (1)](https://github.com/Cyfrin/foundry-full-course-f23/assets/100542180/d1fe45ed-60e8-479c-85eb-0b187ef49093)

**Reply by shikhar229169** [reply]
Ah! you made .env in src folder. Take it out from there.

**Reply by malommey** [reply]
thank you, in which folder should i  create it

**Reply by shikhar229169** [reply]
In your foundry-simple-storage-f23
The same directory in which the .gitignore, foundry.toml are

**Reply by malommey** [reply]
@shikhar229169 like this? 
![Screenshot (2)](https://github.com/Cyfrin/foundry-full-course-f23/assets/100542180/09a94fd5-0722-479d-8789-97ce4ab59c71)

**Reply by shikhar229169** [reply]
No, not like this.
Just make it align with the foundry.toml

**Reply by malommey** [reply]
my src is in the foundry-simple-storage-f23 directory and other files, right? if i open the .env file to  aligned with the foundry.toml you mean i should create the .env file in the foundry-simple-storage-f23 directory, right? which is what i did. 

please what am i missing
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/100542180/77331ab6-76c3-4c94-94f1-fb68791c38c2)

**Reply by shikhar229169** [reply, marked as answer]
Nooo, you created the .env file in the foundry-f23, it should be inside foundry-simple-storage-f23

**Reply by malommey** [reply]
thank you thank you thank you thank you thank you thank you thank it worked

**Reply by 0xKumarShivam** [reply]
thank you Shikhar
